src/neo/Network/P2P/Payloads/Transaction.cs

```diff
@@ -217,7 +247,11 @@ public override int GetHashCode()
 
         public UInt160[] GetScriptHashesForVerifying(StoreView snapshot)
         {
-            var hashes = new HashSet<UInt160>(Cosigners.Keys) { Sender };
+            var hashes = new HashSet<UInt160> { Sender };
+            if (IsOracleResponse)
+                hashes.Add(Blockchain.GetConsensusAddress(NativeContract.Oracle.GetOracleNodes(snapshot)));
+            else
+                hashes.UnionWith(Cosigners.Keys);
```

Comment:
When Oracle nodes construct transactions, they should limit the scope of signatures to prevent users from making malicious transfers in callback functions, right?

Reply:
If it is an oracle response transaction, the cosigners should be the clone of the request transaction.

Reply:
> the cosigners should be the clone of the request transaction.

Will it be better to use the original request transaction or create a new `ScriptContainer` before call `CallFromNativeContract` method.

Reply:
Why?

Reply:
It's not efficient that  response transaction clone the cosginer of the request transaction. The request tx is already on the chain.

Reply:
Updated.